src/parser/mod.rs

```diff
@@ -12012,10 +12033,26 @@ impl<'a> Parser<'a> {
                 replace_into,
                 priority,
                 insert_alias,
+                settings,
+                format_clause,
             }))
         }
     }
 
+    // Parses format clause used for [ClickHouse]. Formats are different when using `SELECT` and
+    // `INSERT` and also when using the CLI for pipes. For `INSERT` it can take an optional values
+    // list which we try to parse here.
+    //
+    // <https://clickhouse.com/docs/en/interfaces/formats>
+    pub fn parse_input_format_clause(&mut self) -> Result<InputFormatClause, ParserError> {
+        let ident = self.parse_identifier()?;
+        let values = self
+            .try_parse(|p| p.parse_comma_separated(|p| p.parse_expr()))
```

Comment:
I think this can use maybe_parse instead? to properly account for unexpected error